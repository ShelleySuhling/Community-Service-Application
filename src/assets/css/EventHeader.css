.events-header{
    display:flex;
    flex-direction:column;
    justify-content: flex-start;
    width: 100%;
    text-align: left;
    font-size: 30px;
    color: #4c92db;
    margin: 0 0 30px 0;
    font-family: 'Lato', sans-serif;
}

.events-header-line{
    display:flex;
    flex-direction: row;
    width: auto;
    flex-wrap: wrap;
}

.dropdown-wrapper{
    width: auto;
    margin: 0 6px 0;
    border-bottom: dotted pink 3px;
}

.dropdown-wrapper.mobile{
    background-color: blue;
    width: auto;
    margin: 0 6px 0;
    border-bottom: dotted pink 3px;
}

.dropdown-wrapper .ui.inline.dropdown .dropdown.icon{
    display: none;
}

