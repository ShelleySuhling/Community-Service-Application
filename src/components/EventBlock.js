import React, { Component } from 'react';
import { Card, Icon } from 'semantic-ui-react'

import moment from 'moment'

class ProfileForm extends Component {

    toDate = (date) => {
        //for whatever reason, firebase's .toDate() method of a timestamp wasn't working so I made my own
        return moment(date.seconds * 1000).format('MM/DD/YYYY')
    }

    toTime = (time) => {
        return moment(time.seconds * 1000).format('hh:mm A')
    }

    generateDescription = () => {
        let { event } = this.props
        return (
            <div className="card-container">
                <div className="card-info-container">
                    <div className="card-title">{event.title}</div>
                    <div className="card-description">
                        Location: {event.location}<br></br>
                        Date: {this.toDate(event.startTime)}<br></br>
                        Time: {this.toTime(event.startTime)} - {this.toTime(event.endTime)}
                    </div>
                </div>
                <div className="card-icons">
                    {event.isMandatory ? <Icon size='huge' color="blue" name='anchor' /> : null}
                </div>
            </div>
        )

    }

    render() {
        let { event } = this.props
        return <Card fluid color='blue' key={event.id} description={this.generateDescription()} />

    }

}

export default ProfileForm