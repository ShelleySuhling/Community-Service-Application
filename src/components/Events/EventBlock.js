import React, { Component } from 'react';
import { Link } from 'react-router-dom'
import { Card, Icon } from 'semantic-ui-react'
import moment from 'moment'

class EventBlock extends Component {

    generateDescription = () => {
        let { event, user } = this.props
        return (
            <div className="card-container">
                <div className="card-info-container">
                    <div className="card-title">
                        {event.title}
                        <div className="card-icons">
                            {/* {event.isMandatory ? <Icon color="blue" name='anchor' /> : null} */}
                        </div>
                    </div>
                    <div className="card-description">
                        <div><Icon color="blue" name='map marker alternate' /> {event.location}</div>
                        <div><Icon color="blue" name='calendar outline' /> {moment(event.startTime).format("MMM Do YY")}</div>
                        <div><Icon color="blue" name='clock outline' />{moment(event.startTime).format('LT')} - {moment(event.endTime).format('LT')}</div>
                    </div>
                </div>
                <div className="card-icon-container">
                    {/* <div className="card-icons">
                    </div> */}
                    {event.isMandatory
                        ? <div className="corner-icon">
                            <Icon size='huge' color="blue" name='caret up' />
                        </div>
                        : <div>{null}</div>}

                    {user.role === "master"
                        ? <Link to={"/edit_event/" + event.id}>
                            <Icon size='large' color="blue" name='pencil alternate' />
                        </Link>
                        : null
                    }

                </div>
            </div>
        )

    }

    render() {
        let { event } = this.props
        return <Card fluid color='blue' key={event.id} description={this.generateDescription()} />

    }

}

export default EventBlock