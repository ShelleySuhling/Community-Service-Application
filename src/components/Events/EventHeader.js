import React, { Component } from 'react';
import { Dropdown } from 'semantic-ui-react'

let context_options = [
    {
        text: "this",
        value: "this"
    },
    {
        text: "next",
        value: "next"
    }
]
let time_options = [
    {
        text: "week",
        value: "week"
    },
    {
        text: "month",
        value: "month"
    }
]

class EventHeader extends Component {

    handleChange = (event, element) => {
        this.props.handleViewChange(element.name, element.value)
    }

    render() {
        let { session } = this.props
        return <div className="events-header">
            <div className="events-header-line">Hey {session.user.fullName.split(' ')[0]},</div>
            <div className="events-header-line">here are your events for
                <div className="dropdown-wrapper">
                    <Dropdown inline options={context_options} name="view_context" defaultValue={context_options[0].value} onChange={this.handleChange} />
                </div>
                <div className="dropdown-wrapper">
                    <Dropdown inline options={time_options} name="view_time" defaultValue={time_options[0].value} onChange={this.handleChange} />
                </div>.
            </div>
        </div>
    }

}

export default EventHeader