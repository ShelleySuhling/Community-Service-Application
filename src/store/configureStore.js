import { applyMiddleware, createStore } from 'redux'
import { persistStore, persistCombineReducers } from 'redux-persist';
import thunkMiddleware from 'redux-thunk'
import storage from 'redux-persist/lib/storage';

import rootReducer from '../reducers/rootReducer'

export default function configureStore(preloadedState) {

    const persistConfig = {
        key: 'root',
        storage: storage,
    };
    const pReducer = persistCombineReducers(persistConfig, rootReducer);

    const store = createStore(pReducer, applyMiddleware(thunkMiddleware));
    const persistor = persistStore(store);
    window.store = store;

    return { persistor, store }
}