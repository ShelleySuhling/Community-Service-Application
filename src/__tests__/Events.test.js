import * as React from 'react';
import { shallow } from 'enzyme';
import moment from 'moment'
import MockDate from 'mockdate'

import Events from '../containers/Events/Events';
import EventBlock from '../components/Events/EventBlock'
import EventHeader from '../components/Events/EventHeader'
import Mobile from '../components/Mobile'
import Default from '../components/Default'

import configureStore from '../store/configureStore';
const { store } = configureStore()

describe('Events', () => {
  MockDate.set('1/1/2018');
  const eventsComponentWrapper = shallow(<Events store={store} />).dive();

  eventsComponentWrapper.setState({
    view_context: "this",
    view_time: "week",
    events: [{
      id: 1,
      title: "first event",
      location: "sample location",
      startTime: moment("2018-01-05 4:30", "YYYY-MM-DD HH:mm"),
      endTime: moment("2018-01-05 7:30", "YYYY-MM-DD HH:mm"),
      isMandatory: true,
    },
    {
      id: 2,
      title: "second event",
      location: "sample location",
      startTime: moment("2018-01-06 4:30", "YYYY-MM-DD HH:mm"),
      endTime: moment("2018-01-06 7:30", "YYYY-MM-DD HH:mm"),
      isMandatory: false,
    },
    {
      id: 3,
      title: "third event",
      location: "sample location",
      startTime: moment("2018-01-10 4:30", "YYYY-MM-DD HH:mm"),
      endTime: moment("2018-01-10 7:30", "YYYY-MM-DD HH:mm"),
      isMandatory: false,
    },
    {
      id: 4,
      title: "fourth event",
      location: "sample location",
      startTime: moment("2018-02-10 4:30", "YYYY-MM-DD HH:mm"),
      endTime: moment("2018-02-10 7:30", "YYYY-MM-DD HH:mm"),
      isMandatory: false,
    }]
  })
  it('matches snapshot', () => {
    expect(eventsComponentWrapper).toMatchSnapshot();
  });
  it('has responsive components', () => {
    expect(eventsComponentWrapper.find(Mobile).getElements()).toHaveLength(1)
    expect(eventsComponentWrapper.find(Default).getElements()).toHaveLength(1)
  })
  it('has 2 EventHeader components', () => {
    expect(eventsComponentWrapper.find(Mobile).find(EventHeader).getElements()).toHaveLength(1)
    expect(eventsComponentWrapper.find(Default).find(EventHeader).getElements()).toHaveLength(1)
  })
  it('has 2 EventBlock components - this week', () => {
    expect(eventsComponentWrapper.find(Mobile).find(EventBlock).getElements()).toHaveLength(2)
    expect(eventsComponentWrapper.find(Default).find(EventBlock).getElements()).toHaveLength(2)
  })
  it('has 4 EventBlock components - next week', () => {
    eventsComponentWrapper.setState({
      view_context: "next",
      view_time: "week",
    })
    expect(eventsComponentWrapper.find(Mobile).find(EventBlock).getElements()).toHaveLength(1)
    expect(eventsComponentWrapper.find(Default).find(EventBlock).getElements()).toHaveLength(1)
  })
  it('has 2 EventBlock components - this month', () => {
    eventsComponentWrapper.setState({
      view_context: "this",
      view_time: "month",
    })
    expect(eventsComponentWrapper.find(Mobile).find(EventBlock).getElements()).toHaveLength(3)
    expect(eventsComponentWrapper.find(Default).find(EventBlock).getElements()).toHaveLength(3)
  })
  it('has 4 EventBlock components - next month', () => {
    eventsComponentWrapper.setState({
      view_context: "next",
      view_time: "month",
    })
    expect(eventsComponentWrapper.find(Mobile).find(EventBlock).getElements()).toHaveLength(1)
    expect(eventsComponentWrapper.find(Default).find(EventBlock).getElements()).toHaveLength(1)
  })
  it('correctly filters view_events - this week', () => {
    eventsComponentWrapper.setState({
      view_context: "this",
      view_time: "week",
    })
    expect(eventsComponentWrapper.instance().collectEventsByView()).toHaveLength(2)
  })
  it('correctly filters view_events - next week', () => {
    eventsComponentWrapper.setState({
      view_context: "next",
      view_time: "week",
    })
    expect(eventsComponentWrapper.instance().collectEventsByView()).toHaveLength(1)
  })
  it('correctly filters view_events - this month', () => {
    eventsComponentWrapper.setState({
      view_context: "this",
      view_time: "month",
    })
    expect(eventsComponentWrapper.instance().collectEventsByView()).toHaveLength(3)
  })
  it('correctly filters view_events - next month', () => {
    eventsComponentWrapper.setState({
      view_context: "next",
      view_time: "month",
    })
    expect(eventsComponentWrapper.instance().collectEventsByView()).toHaveLength(1)
  })
  it('correctly handles view change', () => {
    eventsComponentWrapper.instance().handleViewChange('view_context', 'this')
    eventsComponentWrapper.instance().handleViewChange('view_time', 'week')
    expect(eventsComponentWrapper.state('view_context')).toEqual('this')
    expect(eventsComponentWrapper.state('view_time')).toEqual('week')
  })
});