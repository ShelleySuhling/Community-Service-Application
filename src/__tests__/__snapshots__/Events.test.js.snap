// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Events matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Events
    events={Object {}}
    eventsActions={
      Object {
        "requestCreateEvent": [Function],
        "requestEditEvent": [Function],
        "requestEvents": [Function],
      }
    }
    session={Object {}}
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
    storeSubscription={
      Subscription {
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "unsubscribe": [Function],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Default>
          <div
            className="events-container-desktop"
          >
            <EventHeader
              handleViewChange={[Function]}
              session={Object {}}
            />
            <div
              className="events-card-container"
            >
              <CardGroup>
                <EventBlock
                  event={
                    Object {
                      "endTime": "2018-01-05T15:30:00.000Z",
                      "id": 1,
                      "isMandatory": true,
                      "location": "sample location",
                      "startTime": "2018-01-05T12:30:00.000Z",
                      "title": "first event",
                    }
                  }
                />
                <EventBlock
                  event={
                    Object {
                      "endTime": "2018-01-06T15:30:00.000Z",
                      "id": 2,
                      "isMandatory": false,
                      "location": "sample location",
                      "startTime": "2018-01-06T12:30:00.000Z",
                      "title": "second event",
                    }
                  }
                />
              </CardGroup>
            </div>
          </div>
        </Default>,
        <Mobile>
          <div
            className="events-container-mobile"
          >
            <EventHeader
              handleViewChange={[Function]}
              session={Object {}}
            />
            <CardGroup>
              <EventBlock
                event={
                  Object {
                    "endTime": "2018-01-05T15:30:00.000Z",
                    "id": 1,
                    "isMandatory": true,
                    "location": "sample location",
                    "startTime": "2018-01-05T12:30:00.000Z",
                    "title": "first event",
                  }
                }
              />
              <EventBlock
                event={
                  Object {
                    "endTime": "2018-01-06T15:30:00.000Z",
                    "id": 2,
                    "isMandatory": false,
                    "location": "sample location",
                    "startTime": "2018-01-06T12:30:00.000Z",
                    "title": "second event",
                  }
                }
              />
            </CardGroup>
          </div>
        </Mobile>,
      ],
      "className": "content-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <div
            className="events-container-desktop"
          >
            <EventHeader
              handleViewChange={[Function]}
              session={Object {}}
            />
            <div
              className="events-card-container"
            >
              <CardGroup>
                <EventBlock
                  event={
                    Object {
                      "endTime": "2018-01-05T15:30:00.000Z",
                      "id": 1,
                      "isMandatory": true,
                      "location": "sample location",
                      "startTime": "2018-01-05T12:30:00.000Z",
                      "title": "first event",
                    }
                  }
                />
                <EventBlock
                  event={
                    Object {
                      "endTime": "2018-01-06T15:30:00.000Z",
                      "id": 2,
                      "isMandatory": false,
                      "location": "sample location",
                      "startTime": "2018-01-06T12:30:00.000Z",
                      "title": "second event",
                    }
                  }
                />
              </CardGroup>
            </div>
          </div>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <EventHeader
                handleViewChange={[Function]}
                session={Object {}}
              />,
              <div
                className="events-card-container"
              >
                <CardGroup>
                  <EventBlock
                    event={
                      Object {
                        "endTime": "2018-01-05T15:30:00.000Z",
                        "id": 1,
                        "isMandatory": true,
                        "location": "sample location",
                        "startTime": "2018-01-05T12:30:00.000Z",
                        "title": "first event",
                      }
                    }
                  />
                  <EventBlock
                    event={
                      Object {
                        "endTime": "2018-01-06T15:30:00.000Z",
                        "id": 2,
                        "isMandatory": false,
                        "location": "sample location",
                        "startTime": "2018-01-06T12:30:00.000Z",
                        "title": "second event",
                      }
                    }
                  />
                </CardGroup>
              </div>,
            ],
            "className": "events-container-desktop",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "handleViewChange": [Function],
                "session": Object {},
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <CardGroup>
                  <EventBlock
                    event={
                      Object {
                        "endTime": "2018-01-05T15:30:00.000Z",
                        "id": 1,
                        "isMandatory": true,
                        "location": "sample location",
                        "startTime": "2018-01-05T12:30:00.000Z",
                        "title": "first event",
                      }
                    }
                  />
                  <EventBlock
                    event={
                      Object {
                        "endTime": "2018-01-06T15:30:00.000Z",
                        "id": 2,
                        "isMandatory": false,
                        "location": "sample location",
                        "startTime": "2018-01-06T12:30:00.000Z",
                        "title": "second event",
                      }
                    }
                  />
                </CardGroup>,
                "className": "events-card-container",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": Array [
                    <EventBlock
                      event={
                        Object {
                          "endTime": "2018-01-05T15:30:00.000Z",
                          "id": 1,
                          "isMandatory": true,
                          "location": "sample location",
                          "startTime": "2018-01-05T12:30:00.000Z",
                          "title": "first event",
                        }
                      }
                    />,
                    <EventBlock
                      event={
                        Object {
                          "endTime": "2018-01-06T15:30:00.000Z",
                          "id": 2,
                          "isMandatory": false,
                          "location": "sample location",
                          "startTime": "2018-01-06T12:30:00.000Z",
                          "title": "second event",
                        }
                      }
                    />,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": "first event",
                    "nodeType": "class",
                    "props": Object {
                      "event": Object {
                        "endTime": "2018-01-05T15:30:00.000Z",
                        "id": 1,
                        "isMandatory": true,
                        "location": "sample location",
                        "startTime": "2018-01-05T12:30:00.000Z",
                        "title": "first event",
                      },
                      "user": undefined,
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": "second event",
                    "nodeType": "class",
                    "props": Object {
                      "event": Object {
                        "endTime": "2018-01-06T15:30:00.000Z",
                        "id": 2,
                        "isMandatory": false,
                        "location": "sample location",
                        "startTime": "2018-01-06T12:30:00.000Z",
                        "title": "second event",
                      },
                      "user": undefined,
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                ],
                "type": [Function],
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <div
            className="events-container-mobile"
          >
            <EventHeader
              handleViewChange={[Function]}
              session={Object {}}
            />
            <CardGroup>
              <EventBlock
                event={
                  Object {
                    "endTime": "2018-01-05T15:30:00.000Z",
                    "id": 1,
                    "isMandatory": true,
                    "location": "sample location",
                    "startTime": "2018-01-05T12:30:00.000Z",
                    "title": "first event",
                  }
                }
              />
              <EventBlock
                event={
                  Object {
                    "endTime": "2018-01-06T15:30:00.000Z",
                    "id": 2,
                    "isMandatory": false,
                    "location": "sample location",
                    "startTime": "2018-01-06T12:30:00.000Z",
                    "title": "second event",
                  }
                }
              />
            </CardGroup>
          </div>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <EventHeader
                handleViewChange={[Function]}
                session={Object {}}
              />,
              <CardGroup>
                <EventBlock
                  event={
                    Object {
                      "endTime": "2018-01-05T15:30:00.000Z",
                      "id": 1,
                      "isMandatory": true,
                      "location": "sample location",
                      "startTime": "2018-01-05T12:30:00.000Z",
                      "title": "first event",
                    }
                  }
                />
                <EventBlock
                  event={
                    Object {
                      "endTime": "2018-01-06T15:30:00.000Z",
                      "id": 2,
                      "isMandatory": false,
                      "location": "sample location",
                      "startTime": "2018-01-06T12:30:00.000Z",
                      "title": "second event",
                    }
                  }
                />
              </CardGroup>,
            ],
            "className": "events-container-mobile",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "handleViewChange": [Function],
                "session": Object {},
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": Array [
                  <EventBlock
                    event={
                      Object {
                        "endTime": "2018-01-05T15:30:00.000Z",
                        "id": 1,
                        "isMandatory": true,
                        "location": "sample location",
                        "startTime": "2018-01-05T12:30:00.000Z",
                        "title": "first event",
                      }
                    }
                  />,
                  <EventBlock
                    event={
                      Object {
                        "endTime": "2018-01-06T15:30:00.000Z",
                        "id": 2,
                        "isMandatory": false,
                        "location": "sample location",
                        "startTime": "2018-01-06T12:30:00.000Z",
                        "title": "second event",
                      }
                    }
                  />,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": "first event",
                  "nodeType": "class",
                  "props": Object {
                    "event": Object {
                      "endTime": "2018-01-05T15:30:00.000Z",
                      "id": 1,
                      "isMandatory": true,
                      "location": "sample location",
                      "startTime": "2018-01-05T12:30:00.000Z",
                      "title": "first event",
                    },
                    "user": undefined,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": "second event",
                  "nodeType": "class",
                  "props": Object {
                    "event": Object {
                      "endTime": "2018-01-06T15:30:00.000Z",
                      "id": 2,
                      "isMandatory": false,
                      "location": "sample location",
                      "startTime": "2018-01-06T12:30:00.000Z",
                      "title": "second event",
                    },
                    "user": undefined,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
          ],
          "type": "div",
        },
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Default>
            <div
              className="events-container-desktop"
            >
              <EventHeader
                handleViewChange={[Function]}
                session={Object {}}
              />
              <div
                className="events-card-container"
              >
                <CardGroup>
                  <EventBlock
                    event={
                      Object {
                        "endTime": "2018-01-05T15:30:00.000Z",
                        "id": 1,
                        "isMandatory": true,
                        "location": "sample location",
                        "startTime": "2018-01-05T12:30:00.000Z",
                        "title": "first event",
                      }
                    }
                  />
                  <EventBlock
                    event={
                      Object {
                        "endTime": "2018-01-06T15:30:00.000Z",
                        "id": 2,
                        "isMandatory": false,
                        "location": "sample location",
                        "startTime": "2018-01-06T12:30:00.000Z",
                        "title": "second event",
                      }
                    }
                  />
                </CardGroup>
              </div>
            </div>
          </Default>,
          <Mobile>
            <div
              className="events-container-mobile"
            >
              <EventHeader
                handleViewChange={[Function]}
                session={Object {}}
              />
              <CardGroup>
                <EventBlock
                  event={
                    Object {
                      "endTime": "2018-01-05T15:30:00.000Z",
                      "id": 1,
                      "isMandatory": true,
                      "location": "sample location",
                      "startTime": "2018-01-05T12:30:00.000Z",
                      "title": "first event",
                    }
                  }
                />
                <EventBlock
                  event={
                    Object {
                      "endTime": "2018-01-06T15:30:00.000Z",
                      "id": 2,
                      "isMandatory": false,
                      "location": "sample location",
                      "startTime": "2018-01-06T12:30:00.000Z",
                      "title": "second event",
                    }
                  }
                />
              </CardGroup>
            </div>
          </Mobile>,
        ],
        "className": "content-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <div
              className="events-container-desktop"
            >
              <EventHeader
                handleViewChange={[Function]}
                session={Object {}}
              />
              <div
                className="events-card-container"
              >
                <CardGroup>
                  <EventBlock
                    event={
                      Object {
                        "endTime": "2018-01-05T15:30:00.000Z",
                        "id": 1,
                        "isMandatory": true,
                        "location": "sample location",
                        "startTime": "2018-01-05T12:30:00.000Z",
                        "title": "first event",
                      }
                    }
                  />
                  <EventBlock
                    event={
                      Object {
                        "endTime": "2018-01-06T15:30:00.000Z",
                        "id": 2,
                        "isMandatory": false,
                        "location": "sample location",
                        "startTime": "2018-01-06T12:30:00.000Z",
                        "title": "second event",
                      }
                    }
                  />
                </CardGroup>
              </div>
            </div>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <EventHeader
                  handleViewChange={[Function]}
                  session={Object {}}
                />,
                <div
                  className="events-card-container"
                >
                  <CardGroup>
                    <EventBlock
                      event={
                        Object {
                          "endTime": "2018-01-05T15:30:00.000Z",
                          "id": 1,
                          "isMandatory": true,
                          "location": "sample location",
                          "startTime": "2018-01-05T12:30:00.000Z",
                          "title": "first event",
                        }
                      }
                    />
                    <EventBlock
                      event={
                        Object {
                          "endTime": "2018-01-06T15:30:00.000Z",
                          "id": 2,
                          "isMandatory": false,
                          "location": "sample location",
                          "startTime": "2018-01-06T12:30:00.000Z",
                          "title": "second event",
                        }
                      }
                    />
                  </CardGroup>
                </div>,
              ],
              "className": "events-container-desktop",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "handleViewChange": [Function],
                  "session": Object {},
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <CardGroup>
                    <EventBlock
                      event={
                        Object {
                          "endTime": "2018-01-05T15:30:00.000Z",
                          "id": 1,
                          "isMandatory": true,
                          "location": "sample location",
                          "startTime": "2018-01-05T12:30:00.000Z",
                          "title": "first event",
                        }
                      }
                    />
                    <EventBlock
                      event={
                        Object {
                          "endTime": "2018-01-06T15:30:00.000Z",
                          "id": 2,
                          "isMandatory": false,
                          "location": "sample location",
                          "startTime": "2018-01-06T12:30:00.000Z",
                          "title": "second event",
                        }
                      }
                    />
                  </CardGroup>,
                  "className": "events-card-container",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "children": Array [
                      <EventBlock
                        event={
                          Object {
                            "endTime": "2018-01-05T15:30:00.000Z",
                            "id": 1,
                            "isMandatory": true,
                            "location": "sample location",
                            "startTime": "2018-01-05T12:30:00.000Z",
                            "title": "first event",
                          }
                        }
                      />,
                      <EventBlock
                        event={
                          Object {
                            "endTime": "2018-01-06T15:30:00.000Z",
                            "id": 2,
                            "isMandatory": false,
                            "location": "sample location",
                            "startTime": "2018-01-06T12:30:00.000Z",
                            "title": "second event",
                          }
                        }
                      />,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": "first event",
                      "nodeType": "class",
                      "props": Object {
                        "event": Object {
                          "endTime": "2018-01-05T15:30:00.000Z",
                          "id": 1,
                          "isMandatory": true,
                          "location": "sample location",
                          "startTime": "2018-01-05T12:30:00.000Z",
                          "title": "first event",
                        },
                        "user": undefined,
                      },
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                    Object {
                      "instance": null,
                      "key": "second event",
                      "nodeType": "class",
                      "props": Object {
                        "event": Object {
                          "endTime": "2018-01-06T15:30:00.000Z",
                          "id": 2,
                          "isMandatory": false,
                          "location": "sample location",
                          "startTime": "2018-01-06T12:30:00.000Z",
                          "title": "second event",
                        },
                        "user": undefined,
                      },
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                  ],
                  "type": [Function],
                },
                "type": "div",
              },
            ],
            "type": "div",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <div
              className="events-container-mobile"
            >
              <EventHeader
                handleViewChange={[Function]}
                session={Object {}}
              />
              <CardGroup>
                <EventBlock
                  event={
                    Object {
                      "endTime": "2018-01-05T15:30:00.000Z",
                      "id": 1,
                      "isMandatory": true,
                      "location": "sample location",
                      "startTime": "2018-01-05T12:30:00.000Z",
                      "title": "first event",
                    }
                  }
                />
                <EventBlock
                  event={
                    Object {
                      "endTime": "2018-01-06T15:30:00.000Z",
                      "id": 2,
                      "isMandatory": false,
                      "location": "sample location",
                      "startTime": "2018-01-06T12:30:00.000Z",
                      "title": "second event",
                    }
                  }
                />
              </CardGroup>
            </div>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <EventHeader
                  handleViewChange={[Function]}
                  session={Object {}}
                />,
                <CardGroup>
                  <EventBlock
                    event={
                      Object {
                        "endTime": "2018-01-05T15:30:00.000Z",
                        "id": 1,
                        "isMandatory": true,
                        "location": "sample location",
                        "startTime": "2018-01-05T12:30:00.000Z",
                        "title": "first event",
                      }
                    }
                  />
                  <EventBlock
                    event={
                      Object {
                        "endTime": "2018-01-06T15:30:00.000Z",
                        "id": 2,
                        "isMandatory": false,
                        "location": "sample location",
                        "startTime": "2018-01-06T12:30:00.000Z",
                        "title": "second event",
                      }
                    }
                  />
                </CardGroup>,
              ],
              "className": "events-container-mobile",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "handleViewChange": [Function],
                  "session": Object {},
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": Array [
                    <EventBlock
                      event={
                        Object {
                          "endTime": "2018-01-05T15:30:00.000Z",
                          "id": 1,
                          "isMandatory": true,
                          "location": "sample location",
                          "startTime": "2018-01-05T12:30:00.000Z",
                          "title": "first event",
                        }
                      }
                    />,
                    <EventBlock
                      event={
                        Object {
                          "endTime": "2018-01-06T15:30:00.000Z",
                          "id": 2,
                          "isMandatory": false,
                          "location": "sample location",
                          "startTime": "2018-01-06T12:30:00.000Z",
                          "title": "second event",
                        }
                      }
                    />,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": "first event",
                    "nodeType": "class",
                    "props": Object {
                      "event": Object {
                        "endTime": "2018-01-05T15:30:00.000Z",
                        "id": 1,
                        "isMandatory": true,
                        "location": "sample location",
                        "startTime": "2018-01-05T12:30:00.000Z",
                        "title": "first event",
                      },
                      "user": undefined,
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": "second event",
                    "nodeType": "class",
                    "props": Object {
                      "event": Object {
                        "endTime": "2018-01-06T15:30:00.000Z",
                        "id": 2,
                        "isMandatory": false,
                        "location": "sample location",
                        "startTime": "2018-01-06T12:30:00.000Z",
                        "title": "second event",
                      },
                      "user": undefined,
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                ],
                "type": [Function],
              },
            ],
            "type": "div",
          },
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
